File: database/database-annotation-processor/src/test/java/ru/tinkoff/kora/database/common/annotation/processor/cassandra/repository/AllowedResultsRepository.java
package ru.tinkoff.kora.database.common.annotation.processor.cassandra.repository;

import ru.tinkoff.kora.common.Mapping;
import ru.tinkoff.kora.database.cassandra.CassandraRepository;
import ru.tinkoff.kora.database.common.annotation.Query;
import ru.tinkoff.kora.database.common.annotation.Repository;
import ru.tinkoff.kora.database.common.annotation.processor.cassandra.CassandraEntity;
import ru.tinkoff.kora.database.common.annotation.processor.entity.TestEntityRecord;

import javax.annotation.Nullable;
import java.util.List;
import java.util.Optional;

@Repository
public interface AllowedResultsRepository extends CassandraRepository {
    @Query("INSERT INTO test(test) VALUES ('test')")
    void returnVoid();

    @Query("SELECT test")
    int returnPrimitive();

    @Query("SELECT test")
    Integer returnObject();

    @Query("SELECT test")
    @Nullable
    Integer returnNullableObject();

    @Query("SELECT test")
    @Mapping(CassandraEntity.TestEntityCassandraRowMapper.class)
    TestEntityRecord returnObjectWithRowMapper();

    @Query("SELECT test")
    @Mapping(CassandraEntity.TestEntityCassandraRowMapperNonFinal.class)
    @Nullable
    TestEntityRecord returnObjectWithRowMapperNonFinal();

    @Query("SELECT test")
    @Mapping(CassandraEntity.TestEntityCassandraRowMapper.class)
    Optional<TestEntityRecord> returnOptionalWithRowMapper();

    @Query("SELECT test")
    @Mapping(CassandraEntity.TestEntityCassandraRowMapper.class)
    List<TestEntityRecord> returnListWithRowMapper();
}

File: database/database-symbol-processor/src/test/kotlin/ru/tinkoff/kora/database/symbol/processor/cassandra/repository/AllowedResultsRepository.kt
package ru.tinkoff.kora.database.symbol.processor.cassandra.repository

import ru.tinkoff.kora.common.Mapping
import ru.tinkoff.kora.database.cassandra.CassandraRepository
import ru.tinkoff.kora.database.common.annotation.Query
import ru.tinkoff.kora.database.common.annotation.Repository
import ru.tinkoff.kora.database.symbol.processor.cassandra.TestEntityCassandraRowMapper
import ru.tinkoff.kora.database.symbol.processor.cassandra.TestEntityCassandraRowMapperNonFinal
import ru.tinkoff.kora.database.symbol.processor.entity.TestEntity

@Repository
interface AllowedResultsRepository : CassandraRepository {
    @Query("INSERT INTO test(test) VALUES ('test')")
    fun returnVoid()

    @Query("SELECT test")
    fun returnPrimitive(): Int

    @Query("SELECT test")
    fun returnObject(): Int?

    @Query("SELECT test")
    fun returnNullableObject(): Int?

    @Query("SELECT test")
    @Mapping(TestEntityCassandraRowMapper::class)
    fun returnObjectWithRowMapper(): TestEntity?

    @Query("SELECT test")
    @Mapping(TestEntityCassandraRowMapperNonFinal::class)
    fun returnObjectWithRowMapperNonFinal(): TestEntity?

    @Query("SELECT test")
    @Mapping(TestEntityCassandraRowMapper::class)
    fun returnOptionalWithRowMapper(): TestEntity?

    @Query("SELECT test")
    @Mapping(TestEntityCassandraRowMapper::class)
    fun returnListWithRowMapper(): List<TestEntity>
}
